packages/neos-ui-redux-store/src/CR/Nodes/selectors.ts

```diff
@@ -66,6 +66,67 @@ export const makeGetDocumentNodes = (nodeTypesRegistry: NodeTypesRegistry) => cr
     }
 );
 
+export const makeGetCollapsibleDocumentNodes = (nodeTypesRegistry: NodeTypesRegistry) => createSelector(
+    [
+        nodesByContextPathSelector
+    ],
+    nodesMap => {
+        const documentRole = nodeTypesRegistry.getRole('document');
+        if (!documentRole) {
+            throw new Error('Document role is not loaded!');
+        }
+        const documentSubNodeTypes = nodeTypesRegistry.getSubTypesOf(documentRole);
+
+        const result: NodeMap = {};
+        Object.keys(nodesMap).forEach(contextPath => {
+            const node = nodesMap[contextPath];
+            if (!node) {
+                throw new Error('This error should never be thrown, it\'s a way to fool TypeScript');
+            }
```

Comment:
```suggestion
        Object.entries(nodesMap).forEach(([contextPath, node]) => {
```

This way, there's no need to fool TS :)

Reply:
I tried this, but node can still be undefined in the next line
![image](https://github.com/neos/neos-ui/assets/56877180/31761b86-331a-4d09-bc1d-e1bea73199cb)